quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/DatabasePropertyMappers.java

```diff
@@ -126,7 +130,71 @@ private static String transformDialect(String db, ConfigSourceInterceptorContext
      * For H2 databases we must ensure that the min-pool size is at least one so that the DB is not shutdown until the
      * Agroal connection pool is closed on Keycloak shutdown.
      */
-    private static String transformMinPoolSize(String min, ConfigSourceInterceptorContext context) {
-        return isDevModeDatabase(context) && (min == null || "0".equals(min)) ? "1" : min;
+    private static String transformMinPoolSize(String database, ConfigSourceInterceptorContext context) {
+        return isDevModeDatabase(database) ? "1" : null;
```

Comment:
We might set the min pool size as for the default datasource if not set. I'll add another commit.